File: moderate/trailing-string/trail.pl
open(INFILE, "<$ARGV[0]") or die("Cannot open file $ARGV[0] for reading: $!");
while(chomp(my $line = <INFILE>)) {
    next if($line =~ m/^s$/); # skip blank lines
    ($word, $search) = split(',', $line, 2);
    if($word =~ /$search$/){
        print "1\n";    
    }
    else {
        print "0\n";
    }
}
close(INFILE);

File: moderate/trailing-string/trail.py
import sys
import re
test_cases = open(sys.argv[1], 'r')
for test in test_cases:
    word, search = test.strip().split(',')
    if re.search('{0}$'.format(search), word):
        print 1
    else:
        print 0
test_cases.close()
